### ISSUE TITLE
Improving tokenizer

### ISSUE THREAD

**Issue by aurasphere** [aurasphere/gomorra-sql#15]
So, at the moment GomorraSQL has a very big limitation: the tokenizer.

GomorraSQL executes a query in these steps:

1. the query is split into tokens (i.e. words limited by either a space or a comma). [This is done here by the tokenizer](https://github.com/aurasphere/gomorra-sql/blob/15c2a0fe45c3938b2d5f72c5e3e69c2ed5a197f2/src/main/java/co/aurasphere/gomorrasql/GomorraSqlInterpreter.java#L46) 
2. each token is parsed by the state machine
3. when all token are consumed, the final SQL query is built from the state machine final state

The problem of the current tokenizer is that it fails when we have string parameters that contain either a comma or a white space. Here's an example:

    rifacimm user accunza name accussì "Pinco" -- works
    rifacimm user accunza name accussì "Pinco Pallo" -- doesn't work
    rifacimm user accunza name accussì "Pinco,Pallo" -- doesn't work

The second and third query don't work because they are tokenized in the following way: 
    ['rifacimm', 'user', 'accunza', 'name', 'accussì', '"Pinco', 'Pallo"'] -- actual tokenization
    ['rifacimm', 'user', 'accunza', 'name', 'accussì', '"Pinco Pallo"'] -- what would be a correct tokenization (comma instead of whitespace for third query)

So basically we need to replace the current base tokenizer with one that is aware of quoted strings (paying attention to the keywords that make use of the apostrophe character).

Any help on this is appreciated!

**Comment by matteobaccan** [comment]
I have see this problem when I have implemented my modification
I can works on a new tokenizer that solve this problem and also the problem of empty token that you remove after tokenizer

**Comment by aurasphere** [comment]
Hey @matteobaccan , sure I'll assign the issue to you. What do you mean with "the problem of empty token that you remove after tokenizer"? Thanks!

**Comment by matteobaccan** [comment]
In current tokenizer loop the code remove the empty token after tokenizer. In new tokenizer I wants to have a filtrate output and don't return empty token. 
In this way I think ti have a more clear output.
I'll write some unit test before starting the new tokenizer
